crates/solver/src/driver.rs

```diff
@@ -466,6 +467,20 @@ impl Driver {
                 .collect::<HashSet<_>>() // to avoid duplicates
                 .into_iter()
                 .collect();
+            // external prices for all tokens contained in the trades of a settlement
+            let prices = winning_settlement
+                .settlement
+                .trades()
+                .filter_map(|trade| {
+                    let sell_token = trade.order.data.sell_token;
+                    let buy_token = trade.order.data.buy_token;
+                    Some(vec![
+                        (sell_token, auction_prices.get(&sell_token).cloned()?),
+                        (buy_token, auction_prices.get(&buy_token).cloned()?),
```

Comment:
I think these tokens need to be fallible independently of each other. Meaning that only because the `sell_token_price` can not be found we should **not** automatically discard the `buy_token_price` as well. (and the other way around)

Reply:
In practice, a missing price only happens for a JIT order, where the token prices don't actually matter.

I actually think the more correct here would be to:
```suggestion
                        (sell_token, auction_prices.get(&sell_token).copied().unwrap_or_default()),
                        (buy_token, auction_prices.get(&buy_token).copied().unwrap_or_default()),
```

This way:
1. We guarantee that there is a price for every token in the auction (and we don't need special cases in other places)
2. Tokens that are included as a result of external solver JIT orders have "no value" WRT the auction, which conceptually makes sense to me.

Reply:
> I think these tokens need to be fallible independently of each other.

Since in practice this only affects JIT orders, I _think_ even if we discard both tokens, it won't actually have an impact (I do agree with your point, however, just wanted to point out that there isn't a big practical difference).

Reply:
My initial reasoning is that these prices will be used exclusively for rewards calculation. If one of the prices for a trade is missing, we cant calculate the rewards and there is no point putting the other one anyway.
But I agree we might use this table for other things in the future. Went with @nlordell suggestion, because it is easier to debug why some value is zero than why is it missing (in this case).